pkg/apis/crd/v1alpha1/types.go

```diff
@@ -354,3 +355,148 @@ type BGPPeer struct {
+	// At the moment this is the only supported configuration and it is required for every capture.
+	FirstN *PacketCaptureFirstNConfig `json:"firstN,omitempty"`
+}
+
+// PacketCaptureFileServer specifies the PacketCapture file server information.
+type PacketCaptureFileServer struct {
+	// The URL of the file server. It is set with format: scheme://host[:port][/path],
+	// e.g, https://api.example.com:8443/v1/supportbundles/. If scheme is not set, https is used by default.
+	URL string `json:"url"`
+}
+
+type PacketCaptureSpec struct {
+	// Timeout is the timeout for this capture session. If not specified, defaults to 60s.
+	Timeout       *uint16       `json:"timeout,omitempty"`
+	CaptureConfig CaptureConfig `json:"captureConfig"`
+	// Source is the traffic source we want to perform capture on. Both `Source` and `Destination` is reuqired
+	// for a capture session, and at least one `Pod` should present either in the source or the destination.
+	Source      Source      `json:"source"`
+	Destination Destination `json:"destination"`
+	// Packet defines what kind of traffic we want to capture between the source and desination. If not spicified,
+	// all kinds of traffic will count.
+	Packet *Packet `json:"packet,omitempty"`
+	// FileServer specifies the sftp url config for a file server. If present, captured packets will be uploaded to this server.
+	// If not, the packet capture results will only be present as a file in the antrea-agent container.
+	// When the capture finished, the path information will be shown in `.status.PacketsFilePath`.
+	FileServer *PacketCaptureFileServer `json:"fileServer,omitempty"`
+}
+
+type PacketCaptureStatus struct {
+	// NumCapturedPackets records how many packets have been captured. If it reaches the target number, the capture
+	// can be considered as finished.
+	NumCapturedPackets int32 `json:"numCapturedPackets,omitempty"`
+	// PacketsFilePath is the file path where the captured packets are stored. The format is: "<antrea-agent-pod-name>:<path>".
+	// If `.spec.FileServer` is present, this file will also be uploaded to the target location. This file
+	// will be removed after the PacketCapture CR is deleted.
+	PacketsFilePath string `json:"packetsFilePath"`
+	// StartTime is the time when this capture session started.
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+	// Condition represents the latest available observations of the PacketCapture 's current state.
+	Conditions []PacketCaptureCondition `json:"conditions"`
+}
+
+type PacketCaptureConditionType string
+
+const (
+	// PacketCapturePending means this request is still pending.
+	PacketCapturePending PacketCaptureConditionType = "PacketCapturePedning"
+	// PacketCaptureRunning means antrea is processing this capture request.
+	PacketCaptureRunning PacketCaptureConditionType = "PacketCaptureRunning"
+	// CaptureCompleted means enough packets has been captured and saved in antrea-agent Pod locally already, but results hasn't been
```

Comment:
```suggestion
	// CaptureCompleted means enough packets have been captured and saved in antrea-agent Pod locally already, but results haven't been
```

I think I highlighted this in a previous round of review already

Reply:
OK, will remember this.